plugin/advancedsubscription/src/AdvancedSubscriptionPlugin.php

```diff
@@ -0,0 +1,872 @@
+     * @param int $sessionId
+     * @return array
+     */
+    public function listAllStudentsInQueueBySession($sessionId)
+    {
+        $extraSession = new ExtraFieldValue('session');
+        $session = api_get_session_info($sessionId);
+        $var = $extraSession->get_values_by_handler_and_field_variable($sessionId, 'target');
+        $session['target'] = $var['field_value'];
+        $var = $extraSession->get_values_by_handler_and_field_variable($sessionId, 'publication_end_date');
+        $session['publication_end_date'] = $var['field_value'];
+        $var = $extraSession->get_values_by_handler_and_field_variable($sessionId, 'mode');
+        $session['mode'] = $var['field_value'];
+        $var = $extraSession->get_values_by_handler_and_field_variable($sessionId, 'recommended_number_of_participants');
+        $session['recommended_number_of_participants'] = $var['field_value'];
+        $var = $extraSession->get_values_by_handler_and_field_variable($sessionId, 'vacancies');
+        $session['vacancies'] = $var['field_value'];
+        $queueTable = Database::get_main_table(TABLE_ADV_SUB_QUEUE);
+        $userTable = Database::get_main_table(TABLE_MAIN_USER);
+        $userJoinTable = $queueTable . ' q INNER JOIN ' . $userTable . ' u ON q.user_id = u.user_id';
+        $where = array(
+            'where' =>
+            array(
+                'q.session_id = ? AND q.status <> ? AND q.status <> ?' => array(
+                    $sessionId,
+                    ADV_SUB_QUEUE_STATUS_ADMIN_APPROVED,
+                    ADV_SUB_QUEUE_STATUS_ADMIN_DISAPPROVED,
+                )
+            )
+        );
+        $select = 'u.user_id, u.firstname, u.lastname, q.created_at, q.updated_at, q.status, q.id as queue_id';
+        $students = Database::select($select, $userJoinTable, $where);
+        foreach ($students as &$student) {
+            $status = intval($student['status']);
+            switch($status) {
+                case ADV_SUB_QUEUE_STATUS_NO_QUEUE:
+                case ADV_SUB_QUEUE_STATUS_START:
+                    $student['validation'] = '';
+                    break;
+                case ADV_SUB_QUEUE_STATUS_BOSS_DISAPPROVED:
+                case ADV_SUB_QUEUE_STATUS_ADMIN_DISAPPROVED:
+                    $student['validation'] = get_lang('No');
+                    break;
+                case ADV_SUB_QUEUE_STATUS_BOSS_APPROVED:
+                case ADV_SUB_QUEUE_STATUS_ADMIN_APPROVED:
+                    $student['validation'] = get_lang('Yes');
+                    break;
+                default:
+                    error_log(__FILE__ . ' ' . __FUNCTION__ . ' Student status no detected');
+            }
+        }
+        $return = array(
+            'session' => $session,
+            'students' => $students,
+        );
+
+        return $return;
+
+    }
+
+    /**
+     * List all session (id, name) for select input
+     * @return array
+     */
+    public function listAllSessions()
```

Comment:
There should be a default LIMIT parameter to this function. Think about cases with 500,000 sessions